tests/unit/s2n_client_key_share_extension_test.c

```diff
@@ -938,319 +941,974 @@ int main(int argc, char **argv)
-
-        if (s2n_is_in_fips_mode()) {
-            /* Test that s2n_client_key_share_extension.send sends only ECC key shares when in FIPS mode,
-             * even if tls13_kem_groups is non-null. */
-            const struct s2n_kem_preferences test_kem_prefs = {
+        const struct s2n_kem_preferences kem_prefs_all = {
                 .kem_count = 0,
                 .kems = NULL,
                 .tls13_kem_group_count = s2n_array_len(all_kem_groups),
                 .tls13_kem_groups = all_kem_groups,
-            };
+        };
 
-            const struct s2n_security_policy test_security_policy = {
+        const struct s2n_security_policy security_policy_all = {
                 .minimum_protocol_version = S2N_SSLv3,
                 .cipher_preferences = &cipher_preferences_test_all_tls13,
-                .kem_preferences = &test_kem_prefs,
+                .kem_preferences = &kem_prefs_all,
                 .signature_preferences = &s2n_signature_preferences_20200207,
                 .ecc_preferences = &s2n_ecc_preferences_20200310,
-            };
+        };
 
-            struct s2n_connection *conn;
-            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
-            conn->security_policy_override = &test_security_policy;
+        const struct s2n_kem_group *kem_groups_sike[] = {
+                &s2n_secp256r1_sike_p434_r2,
+        };
+        const struct s2n_kem_preferences kem_prefs_sike = {
+                .kem_count = 0,
+                .kems = NULL,
+                .tls13_kem_group_count = s2n_array_len(kem_groups_sike),
+                .tls13_kem_groups = kem_groups_sike,
+        };
 
-            const struct s2n_kem_preferences *kem_pref = NULL;
-            EXPECT_SUCCESS(s2n_connection_get_kem_preferences(conn, &kem_pref));
-            EXPECT_NOT_NULL(kem_pref);
-            EXPECT_EQUAL(kem_pref->tls13_kem_group_count, S2N_SUPPORTED_KEM_GROUPS_COUNT);
+        const struct s2n_security_policy security_policy_sike = {
+                .minimum_protocol_version = S2N_SSLv3,
+                .cipher_preferences = &cipher_preferences_test_all_tls13,
+                .kem_preferences = &kem_prefs_sike,
+                .signature_preferences = &s2n_signature_preferences_20200207,
+                .ecc_preferences = &s2n_ecc_preferences_20200310,
+        };
 
-            const struct s2n_ecc_preferences *ecc_preferences = NULL;
-            EXPECT_SUCCESS(s2n_connection_get_ecc_preferences(conn, &ecc_preferences));
-            EXPECT_NOT_NULL(ecc_preferences);
+        EXPECT_EQUAL(S2N_SUPPORTED_KEM_GROUPS_COUNT, s2n_array_len(all_kem_groups));
 
-            DEFER_CLEANUP(struct s2n_stuffer key_share_extension = { 0 }, s2n_stuffer_free);
-            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&key_share_extension, 1024));
-            EXPECT_SUCCESS(s2n_client_key_share_extension.send(conn, &key_share_extension));
+        /* Tests for s2n_client_key_share_extension.send */
```

Comment:
There were no material changes to these `s2n_client_key_share_extension.send` tests. The git diff here is deceptive. 

All the new tests that were added for `s2n_client_key_share_extension.recv` begin on line 1264.